extensions/vscode/src/granite/utils/compatibilityUtils.ts

```diff
@@ -2,16 +2,55 @@ import * as vscode from "vscode";
 
 import type { VsCodeWebviewProtocol } from "../../webviewProtocol";
 
-const incompatibleExtensionIds = new Set(["Continue.continue"]);
-export function checkForIncompatibleExtensions(): boolean {
+export type ExtensionInfo = {
+  id: string;
+  name: string;
+};
+
+export type ExtensionConflictReport = {
+  currentExtension: ExtensionInfo;
+  conflictingExtensions: ExtensionInfo[];
+};
+
+const extensionNameLookup = new Map<string, string>([
+  ["redhat.granitecode" /* no-transform */, "Granite.Code" /* no-transform */],
+  ["Continue.continue" /* no-transform */, "Continue" /* no-transform */],
+]);
+
+// We have to omit this file while building, otherwise it would not work properly
+const incompatibleExtensionIds = new Set(extensionNameLookup.keys())
+let currentExtensionId = "";
+
+export function checkForIncompatibleExtensions(): ExtensionConflictReport {
   const extensions = vscode.extensions.all;
-  return extensions.some((e) => incompatibleExtensionIds.has(e.id));
+  const conflictingExtensions: ExtensionInfo[] = extensions
+    .filter((e) => incompatibleExtensionIds.has(e.id))
+    .map((e) => {
+      return {
+        id: e.id,
+        name: extensionNameLookup.get(e.id)!,
+      };
+    });
+  return {
+    currentExtension: {
+      id: currentExtensionId,
+      name: extensionNameLookup.get(currentExtensionId)!,
+    },
+    conflictingExtensions,
+  };
 }
 
 export function setupExtensionCheck(
   context: vscode.ExtensionContext,
   webviewMessenger: VsCodeWebviewProtocol,
 ) {
+  // Grab current extension id during activation
+  if (currentExtensionId.length === 0) {
```

Comment:
setupExtensionCheck is only ever called once at startup, you don't need that if statement, just set currentExtensionId